crates/matrix-sdk/tests/integration/room/thread.rs

```diff
@@ -0,0 +1,57 @@
+use matrix_sdk::test_utils::mocks::MatrixMockServer;
+use matrix_sdk_test::async_test;
+use ruma::{owned_event_id, room_id};
+
+#[async_test]
+async fn test_subscribe_thread() {
+    let server = MatrixMockServer::new().await;
+    let client = server.client_builder().build().await;
+
+    let room_id = room_id!("!test:example.org");
+    let room = server.sync_joined_room(&client, room_id).await;
+
+    let root_id = owned_event_id!("$root");
+
+    server
+        .mock_put_thread_subscription()
+        .match_room_id(room_id.to_owned())
+        .match_thread_id(root_id.clone())
+        .ok()
+        .mock_once()
+        .mount()
+        .await;
+
+    // I can subscribe to a thread.
+    room.subscribe_thread(root_id.clone(), true).await.unwrap();
+
+    server
+        .mock_get_thread_subscription()
+        .match_room_id(room_id.to_owned())
+        .match_thread_id(root_id.clone())
+        .ok(true)
+        .mock_once()
```

Comment:
I see a `mock_once` here and then two calls to `fetch_thread_subscription` - is this OK?